lib/osf-components/addon/components/make-decision-dropdown/component.ts

```diff
@@ -0,0 +1,258 @@
+interface Args {
+    registration: RegistrationModel;
+    collectionSubmission: CollectionSubmissionModel;
+}
+
+export default class MakeDecisionDropdown extends Component<Args> {
+    @service intl!: Intl;
+    @service store!: Store;
+    @service toast!: Toast;
+    @service router!: RouterService;
+
+    @tracked decisionTrigger?: AllActionTriggers;
+    @tracked comment?: string;
+
+    actionTriggerToLabelKey: {};
+    actionTriggerToDescriptionKey: {};
+
+    constructor(owner: unknown, args: Args) {
+        super(owner, args);
+        assert('Make decision dropdown requires a registration xor collection-submission',
+            Boolean(args.registration) !== Boolean(args.collectionSubmission));
+        if (args.registration) {
+            if (this.revisionIsPending) {
+                this.actionTriggerToLabelKey = SchemaResponseActionTriggerToLabelMapKey;
+                this.actionTriggerToDescriptionKey = SchemaResponseActionTriggerToDescriptionKey;
+            } else {
+                this.actionTriggerToLabelKey = ReviewActionTriggerToTextLabelKey;
+                this.actionTriggerToDescriptionKey = ReviewActionTriggerToDescriptionKey;
+            }
+        } else {
+            this.actionTriggerToLabelKey = CollectionSubmissionActionTriggerToLabelMapKey;
+            this.actionTriggerToDescriptionKey = CollectionSubmissionActionTriggerToDescriptionKey;
+        }
+    }
+
+    get latestRevision() {
+        return this.args.registration.schemaResponses.firstObject;
+    }
+
+    get revisionIsPending() {
+        if (this.args.registration) {
+            return (this.args.registration.reviewsState === RegistrationReviewStates.Accepted
+            || this.args.registration.reviewsState === RegistrationReviewStates.Embargo)
+            && this.args.registration.revisionState === RevisionReviewStates.RevisionPendingModeration;
+        }
+        return false;
+    }
+
+    get commentTextArea() {
+        if (this.args.registration) {
+            if (this.args.registration.reviewsState) {
+                if ([RegistrationReviewStates.Pending, RegistrationReviewStates.PendingWithdraw]
+                    .includes(this.args.registration.reviewsState) ||
+                    this.revisionIsPending) {
+                    return {
+                        label: this.intl.t('osf-components.makeDecisionDropdown.additionalComment'),
+                        placeholder: this.intl.t('osf-components.makeDecisionDropdown.additionalCommentPlaceholder',
+                            { targetType: this.intl.t('osf-components.makeDecisionDropdown.registration') }),
+                    };
+                }
+
+                return {
+                    label: this.intl.t('osf-components.makeDecisionDropdown.justificationForWithdrawal'),
+                    placeholder:
+                        this.intl.t('osf-components.makeDecisionDropdown.justificationForWithdrawalPlaceholder'),
+                };
```

Comment:
This might be worth breaking into a function for better readability. The nesting gets a little deep here.